app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CommentsController < ApplicationController
+  before_action :set_company, only: [:index, :new, :create, :edit, :update, :destroy]
+  before_action :set_job, only: [:index, :new, :create, :edit, :update, :destroy]
+  before_action :set_comment, only: [:edit, :update, :destroy]
+
+  def index
+    render('jobs/show')
+  end
+
+  def new
+    @comment = @job.comments.new
+  end
+
+  def create
+    @comment = @job.comments.create(comment_params)
+    if @comment.save
+      flash[:success] = "You created a comment for the #{@job.title} job at #{@company.name}."
+      redirect_to company_job_comment_path(@company, @job, @comment)
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @comment.update(comment_params)
+      redirect_to company_job_comment_path(@company, @job, @comment)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @comment.delete
+    redirect_to company_job_path(@company, @job)
+  end
+
+  def set_comment
```

Comment:
since you are only using these methods within this class in your before filters, i suggest making these private